tensorboard/plugins/interactive_inference/witwidget/notebook/visualization.py

```diff
@@ -538,3 +576,47 @@ def set_target_feature(self, target):
   def _set_uses_json_input(self, is_json):
     self.store('uses_json_input', is_json)
     return self
+
+  def _set_uses_json_list(self, is_list):
+    self.store('uses_json_list', is_list)
+    return self
+
+  def set_feature_names(self, feature_names):
+    """Sets the feature names of the provided list-based examples.
+
+    If examples are provided in JSON as lists, then this method can provide
+    a string feature name for each value in the example lists for readability.
+
+    Args:
+      feature_names: A list of strings where each index is the feature name of
+      the value at that index in each example's value list, for display
+      purposes.
+
+    Returns:
+      self, in order to enabled method chaining.
+    """
+    self.store('feature_names', feature_names)
+    # If examples have already been stored, update them to now use these
+    # provided feature names.
+    if self.config.get('examples'):
+      self.set_examples(self._add_feature_names_to_examples())
+    return self
+
+  def _add_feature_names_to_examples(self):
```

Comment:
Does this fail if I call set_feature_names twice as int(feat) is now casting feature name string to an integer?

Reply:
after discussion, made this logic simpler by having feature_names optionally provided with examples, so no need to ever convert existing tf.Examples to have the new feature name mapping. its all done at set_examples time now.